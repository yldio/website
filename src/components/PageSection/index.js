import Container from 'components/Container';
import { spacing } from 'styles/spacing';

const PageSection = Container.withComponent('section').extend`
  margin-bottom: ${spacing(10)};
  margin-top: ${spacing(10)};
  position: relative;

  ${props =>
    props.gradient &&
    `
    padding-bottom: ${spacing(10)};
    &:after {
      content: '';
      position: absolute;
      bottom: 0;
      left: -7.5%;
      height: 50%;
      width: 115%;
      background-image: linear-gradient(
        -180deg,
        rgba(255, 255, 255, 0.5) 0%,
        #f3f5fa 52%
      );
      z-index: -1;
    }
  `}
`;

export default PageSection;
