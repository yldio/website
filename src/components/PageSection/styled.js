import Container from 'components/Container';
import { spacing } from 'styles/spacing';

export const Section = Container.withComponent('section').extend`
  margin-bottom: ${spacing(10)};
  margin-top: ${spacing(10)};
  position: relative;

  padding-bottom: 0;
`;
