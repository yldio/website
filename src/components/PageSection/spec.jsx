import React from 'react';
import { shallow } from 'enzyme';
import PageSection from './index';

test('snapshot', () => {
  const wrapper = shallow(<PageSection />);

  expect(wrapper).toMatchSnapshot();
});

test('snapshot with gradient', () => {
  const wrapper = shallow(<PageSection gradient>Content</PageSection>);

  expect(wrapper).toMatchSnapshot();
});
