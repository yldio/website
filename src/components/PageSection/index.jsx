import React from 'react';
import PropTypes from 'prop-types';

import { Section } from './styled';

const PageSection = ({ children, ...props }) => (
  <Section {...props}>{children}</Section>
);

PageSection.propTypes = {
  children: PropTypes.node
};

PageSection.defaultProps = {
  children: []
};

export default PageSection;
