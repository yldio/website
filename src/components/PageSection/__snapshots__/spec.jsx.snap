// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `<Container />`;

exports[`snapshot with gradient 1`] = `
<Container
  gradient={true}
>
  Content
</Container>
`;
